vendor/github.com/openshift/hive/apis/hive/v1/aws/platform.go

```diff
@@ -17,4 +17,24 @@ type Platform struct {
 	// UserTags specifies additional tags for AWS resources created for the cluster.
 	// +optional
 	UserTags map[string]string `json:"userTags,omitempty"`
+
+	// STS specifies configuration for deploying a cluster that uses the AWS Security Token Service, instead of long lived credentials.
+	// +optional
+	STS *STS `json:"sts,omitempty"`
+}
+
+// STS specifies configuration for deploying a cluster that uses the AWS Security Token Service, instead of long lived credentials.
+type STS struct {
+	// ServiceAccountIssuerKeySecretRef refers to a secret that contains a 'bound-service-account-signing-key.key' data key pointing to the private key that will be used to sign ServiceAccount objects.
+	ServiceAccountIssuerKeySecretRef corev1.LocalObjectReference `json:"serviceAccountIssuerKeySecretRef"`
+
+	CredentialsRequestRoles []CredentialsRequestRoles `json:"credentialsRequestRoles"`
```

Comment:
Similar comment to the one about having the signing key be part of the API. What if the Hive API had a .spec.preInstallManifests[] that was just filename/path with yaml/json that would be inserted into the install pod at the appropriate time before calling 'create cluster'?
Or maybe even just a single Secret where each key/value pair was the file and the content to insert...